iceoryx_posh/test/moduletests/test_popo_server_port_user.cpp

```diff
@@ -0,0 +1,732 @@
+
+    EXPECT_TRUE(sut.portUser.hasLostRequestsSinceLastCall());
+}
+
+TEST_F(ServerPort_test, HasLostRequestsSinceLastCallWhenNoFurtherRequestAreLostReturnsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "27a8ddcc-e22b-4bd9-abbd-ed61ac88d01c");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    pushRequests(sut.requestQueuePusher, QUEUE_CAPACITY + 1);
+    IOX_DISCARD_RESULT(sut.portUser.hasLostRequestsSinceLastCall());
+
+    EXPECT_FALSE(sut.portUser.hasLostRequestsSinceLastCall());
+}
+
+TEST_F(ServerPort_test, HasLostRequestsSinceLastCallWhenFurtherRequestAreLostReturnsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "21fd5e5e-93e4-4ef9-8448-d1c6ae6e3989");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    pushRequests(sut.requestQueuePusher, QUEUE_CAPACITY + 1);
+    IOX_DISCARD_RESULT(sut.portUser.hasLostRequestsSinceLastCall());
+    pushRequests(sut.requestQueuePusher, 1);
+
+    EXPECT_TRUE(sut.portUser.hasLostRequestsSinceLastCall());
+}
+
+TEST_F(ServerPort_test, HasLostRequestsSinceLastCallWhenNoRequestAreLostAfterRemovingRequestFromQueueReturnsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "248ddc67-5717-4b33-8b04-80d8246fedb3");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    pushRequests(sut.requestQueuePusher, QUEUE_CAPACITY + 1);
+    IOX_DISCARD_RESULT(sut.portUser.hasLostRequestsSinceLastCall());
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_FALSE(sut.portUser.hasLostRequestsSinceLastCall());
+}
+
+// END hasLostRequestsSinceLastCall tests
+
+// BEGIN allocateResponse tests
+
+TEST_F(ServerPort_test, AllocateResponseWithNullptrAsRequestHeaderCallsErrorHandler)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "05891e1b-3aa4-4aa9-aafd-0649f88d2982");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    iox::cxx::optional<iox::Error> detectedError;
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+            EXPECT_THAT(error, Eq(iox::Error::kEXPECTS_ENSURES_FAILED));
+            EXPECT_THAT(errorLevel, Eq(iox::ErrorLevel::FATAL));
+            detectedError.emplace(error);
+        });
+
+    constexpr uint64_t USER_PAYLOAD_SIZE{8U};
+    constexpr uint64_t USER_PAYLOAD_ALIGNMENT{8U};
+
+    IOX_DISCARD_RESULT(sut.portUser.allocateResponse(nullptr, USER_PAYLOAD_SIZE, USER_PAYLOAD_ALIGNMENT));
+
+    EXPECT_TRUE(detectedError.has_value());
+}
+
+TEST_F(ServerPort_test,
+       AllocateResponseWithInvalidPayloadParameterReturns_INVALID_PARAMETER_FOR_USER_PAYLOAD_OR_USER_HEADER)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6015d3d6-73ae-46fe-994d-445f3f8f366d");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t USER_PAYLOAD_SIZE{23U};
+    constexpr uint64_t USER_PAYLOAD_ALIGNMENT{15U};
```

Comment:
Am I right that here the payload alignment is the only invalid parameter. Could you please rename it into `USER_INVALID_PAYLOAD_ALIGNMENT`.

Reply:
The payload size is also invalid since it must be a multiple of the alignment

Reply:
Oh I didn't know that this was required. Could you then please add to both variables `INVALID`.